python/ray/autoscaler/v2/scheduler.py

```diff
@@ -468,7 +468,22 @@ def _try_schedule_one(
             True if the resource request is scheduled on this node.
         """
 
-        # TODO: add labels when implementing the gang scheduling.
+        # Check if there's placement constraints that are not satisfied.
+        for constraint in request.placement_constraints:
+            if constraint.HasField("anti_affinity"):
+                anti_affinity = constraint.anti_affinity
+                if (
+                    anti_affinity.label_name in self.labels
+                    and anti_affinity.label_value
+                    == self.labels[anti_affinity.label_name]
+                ):
+                    # The node already has a label that matches the anti-affinity
+                    return False
+
+            # We don't need to check for affinity constraints here since
+            # affinity doesn't affect if a request can be scheduled on a node.
+            # Affinity constraints are only used for scoring.
```

Comment:
Mention that the bundles are combined into one and STRICT_PACK doesn't have partial recovery.